### DISCUSSION TITLE
Did it available to use when an issue labeled?

### DISCUSSION THREAD

**Discussion by ghost** [stefanzweifel/git-auto-commit-action#272] [Help] [answered]
```yaml
on:
  issues:
    types: [labeled]
```

Code: https://github.com/aoeer39/acmerindex.github.io/blob/02c1b750baf652defcd74734b9c5bb32664f3b60/.github/workflows/new_group.yml

How to fix the following issue? 

```
INPUT_REPOSITORY value: .
INPUT_STATUS_OPTIONS: 
INPUT_FILE_PATTERN: .
INPUT_BRANCH value: main
Already on 'main'
M   data/groups.yaml
Your branch is up to date with 'origin/main'.
INPUT_ADD_OPTIONS: 
INPUT_FILE_PATTERN: .
INPUT_COMMIT_OPTIONS: 
INPUT_COMMIT_USER_NAME: github-actions[bot]
INPUT_COMMIT_USER_EMAIL: 41898282+github-actions[bot]@users.noreply.github.com
INPUT_COMMIT_MESSAGE: Update groups.yaml (Closes #2)
INPUT_COMMIT_AUTHOR: aoeer39 <aoeer39@users.noreply.github.com>
[main 66acd4c] Update groups.yaml (Closes #2)
Author: aoeer39 <aoeer39@users.noreply.github.com>
1 file changed, 4 insertions(+)
INPUT_TAGGING_MESSAGE: 
No tagging message supplied. No tag will be added.
INPUT_PUSH_OPTIONS: 
fatal: could not read Username for 'https://github.com/': No such device or address
Error: Invalid status code: 128
at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
at ChildProcess.emit (node:events:390:28)
at maybeClose (node:internal/child_process:1064:16)
at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {
code: 128
}
Error: Invalid status code: 128
at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
at ChildProcess.emit (node:events:390:28)
at maybeClose (node:internal/child_process:1064:16)
at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
```

**Comment by stefanzweifel** [comment, marked as answer]
_I've updated your log output so it's clearer to see, what is actually going on._

---

The actual error is this:

> fatal: could not read Username for 'https://github.com/': No such device or address

As you listen to the `issues`-event, you must tell GitHub Actions which branch should be checked out.
Update your workflow to add a `ref`-argument to `actions/checkout`:

```diff
      - name: Checkout
        uses: actions/checkout@v3
        with:
+         ref: main
          persist-credentials: false
          fetch-depth: 0
```

See also https://github.com/stefanzweifel/git-auto-commit-action#checkout-the-correct-branch.

---

Note: Using git-auto-commit with the `issues`-event is not encouraged or tested. It can well be, that the Action does not work at all.

**Comment by hugoovf** [comment]
I've the same error, I want to commit on a pull request.
![image](https://user-images.githubusercontent.com/69076992/214119517-1b13c7f9-a215-4312-a353-1cbb407f3bb0.png)
And I've this already:
- uses: actions/checkout@v2
  with:
    ref: ${{ github.head_ref }}

**Reply by stefanzweifel** [reply]
@hugovidafe  Hm, you'r error is:

> fatal: could not read Username for github.com: No such device or address exists.

Could you please share your entire workflow? From the screenshot alone I can't figure out what the issue is.

**Reply by hugoovf** [reply]
> Could you please share your entire workflow? From the screenshot alone I can't figure out what the issue is.

https://github.com/hugovidafe/hugovidafe/actions/runs/3989152822/workflow

Link updated!

**Reply by stefanzweifel** [reply]
Wow. What a workflow.
I can't help you directly, but I can guide you in the right direction.

First: `ref` is not a valid input for `git-auto-commit`. GitHub Actions warns about this here: https://github.com/hugovidafe/hugovidafe/actions/runs/3989312753.
You probably want to use `branch`.

Second: the problem is that you don't call `actions/checkout` in the `release` step. 
`actions/checkout` sets up the authentication between the git repository and GitHub.
If `actions/checkout` is missing, `git-auto-commit` can't push to GitHub. (Not even sure the step is within a Git repository).

> **Note**
> Each step in GitHub Actions is isolated from all the other steps.
> Calling `actions/checkout` in your "dependencies"-step does't make the git repository of your project not automatically available to all other steps.

I'm not sure how one could add a checkout step here, so that it does what you want to achieve here.

```diff
release:
    name: 🚀 Release
    runs-on: ubuntu-latest
    needs: pull_request

    steps:

+     - name: ⬇️ Checkout repository
+       uses: actions/checkout@main
+       with:
+         fetch-depth: 0
+         ref: ${{ github.head_ref }}

      - name: ⤵️ Get build
        uses: actions/download-artifact@main
        with:
          name: full_build

      - name: 💽 Setup Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 'lts/*'

      - name: 📦 Untar build
        run: tar -xzf build.tar.gz && rm -rf build.tar.gz

      # # Get the labels with github.event.pull_request.labels.*.name
      # # and filter labels to get major, minor, patch or prerelease and save it in the environment as VERS_ENUM
      # - name: 🔎 Filter labels
      #   shell: bash
      #   run: echo "VERS_ENUM=$(node -p -e "const labels = '${{ toJson(github.event.pull_request.labels) }}'; if (labels.includes('major')) 'major'; else if (labels.includes('minor')) 'minor'; else if (labels.includes('patch')) 'patch'; else 'prerelease';")" >> $GITHUB_ENV

      # - name: 🔖 Bump version
      #   run: |
      #     yarn workspaces foreach version ${{ env.VERS_ENUM }}
      #     echo "VERSION=$(node -p -e "require('./package.json').version")" >> $GITHUB_ENV

      - name: 🧨 Experiment
        run: echo "Experiment" >> experiment.txt

      - name: ⬆️ Commit changes
        uses: stefanzweifel/git-auto-commit-action@v4
        with:
          ref: ${{ github.head_ref }}
          commit_message: 🔖 release v${{ env.VERSION }}
          commit_options: '-n'
          tagging_message: 'v${{ env.VERSION }}'
```

Last: you're creating a git tag and push that tag to the branch of the pull request. If the branch is deleted, the tag is gone too (I think).
I would suggest pushing your tags to your default branch (`main`).

**Reply by hugoovf** [reply]
> > Note
> > Each step in GitHub Actions is isolated from all the other steps.
> > Calling `actions/checkout` in your "dependencies"-step does't make the git repository of your project not automatically available to all other steps.
> 
> I'm not sure how one could add a checkout step here, so that it does what you want to achieve here.
> 
> ```diff
> release:
>     name: 🚀 Release
>     runs-on: ubuntu-latest
>     needs: pull_request
> 
>     steps:
> 
> +     - name: ⬇️ Checkout repository
> +       uses: actions/checkout@main
> +       with:
> +         fetch-depth: 0
> +         ref: ${{ github.head_ref }}
> ```

I tried this and I'm getting the same error.
https://github.com/hugovidafe/hugovidafe/actions/runs/3989761515/jobs/6842625930#step:7:28
Get the reference from Workflow file of the left menu ^^

**Reply by stefanzweifel** [reply]
@hugovidafe Yeah, this wasn't mean as **the solution** that magically fixes your problem. :)
I think the problem now is, that your "Get Build" step overrides the repository, that has been checked out a step before.

You somehow have to

- checkout the repository
- download your artifact and extract it into a new directory
- move the files that need to be tracked by Git into the folder with the git repository
- execute the code that changes stuff
- run git-auto-commit
